File: CSharp/Parrot/AfricanParrot.cs
using System;

namespace Parrot;

public class AfricanParrot : Parrot
{
    protected int _numberOfCoconuts;

    public AfricanParrot(int numberOfCoconuts)
    {
        _numberOfCoconuts = numberOfCoconuts;
    }

    public override double GetSpeed()
    {
        return Math.Max(0, GetBaseSpeed() - GetLoadFactor() * _numberOfCoconuts);
    }

    public override string GetCry()
    {
        return "Sqaark!";
    }
}

File: Java/src/main/com/parrot/AfricanParrot.java
package com.parrot;

public class AfricanParrot extends Parrot {

    public AfricanParrot(int numberOfCoconuts, double voltage, boolean isNailed) {
        super( numberOfCoconuts, voltage, isNailed);
    }

    @Override
    public double getSpeed() {
        return Math.max(0, BASE_SPEED - LOAD_FACTOR * numberOfCoconuts );
    }

    @Override
    protected String getCry() {
        return "Sqaark!";
    }
}
